code/datums/components/anti_magic.dm

```diff
@@ -15,12 +15,43 @@
 	holy = _holy
 
 /datum/component/anti_magic/proc/on_equip(datum/source, mob/equipper, slot)
+	if(slot == SLOT_IN_BACKPACK)
+		return
 	RegisterSignal(equipper, COMSIG_MOB_RECEIVE_MAGIC, .proc/can_protect, TRUE)
 
 /datum/component/anti_magic/proc/on_drop(datum/source, mob/user)
 	UnregisterSignal(user, COMSIG_MOB_RECEIVE_MAGIC)
 
-/datum/component/anti_magic/proc/can_protect(datum/source, _magic, _holy, list/protection_sources)
+/datum/component/anti_magic/proc/can_protect(datum/source, _magic, _holy, major, list/protection_sources)
 	if((_magic && magic) || (_holy && holy))
 		protection_sources += parent
+		react(major)
 		return COMPONENT_BLOCK_MAGIC
+		
+/datum/component/anti_magic/proc/react(major)
+	return
+		
+		
+/datum/component/anti_magic/holymelon
```

Comment:
Don't make a subtype for this, add in a capacity to have uses on the regular component

Reply:
i would also have to split between antimagic that deletes the item and antimagic that deletes itself, and it would start to bloat the arg list for the component's initialize i think